backend/LexBoxApi/GraphQL/OrgMutations.cs

```diff
@@ -89,6 +89,41 @@ public async Task<IQueryable<Organization>> AddProjectToOrg(
         return dbContext.Orgs.Where(o => o.Id == orgId);
     }
 
+    [Error<DbError>]
+    [Error<NotFoundException>]
+    [UseMutationConvention]
+    [UseFirstOrDefault]
+    [UseProjection]
+    public async Task<IQueryable<Organization>> AddProjectsToOrg(
+        LexBoxDbContext dbContext,
+        IPermissionService permissionService,
+        [Service] ProjectService projectService,
+        Guid orgId,
+        IEnumerable<Guid> projectIds)
+    {
+        var org = await dbContext.Orgs.Include(o => o.Members).Include(o => o.Projects).SingleOrDefaultAsync(o => o.Id == orgId);
+        NotFoundException.ThrowIfNull(org);
+        permissionService.AssertCanAddProjectToOrg(org);
+        // First make sure ALL projects pass permissions tests
+        foreach (var projectId in projectIds)
+        {
+            await permissionService.AssertCanManageProject(projectId);
+        }
+        // Now exclude any projects that don't actually exist or the org already has or that don't exist
+        var alreadyInOrg = org.Projects.Select(o => o.Id).ToHashSet();
+        var filteredIds = projectIds.Where(id => !alreadyInOrg.Contains(id));
```

Comment:
same here, this enumerable is iterated multiple times, meaning the where will be evaluated multiple times

Reply:
Done, converted to a HashSet since it's being looked up via `.Contains` later on, but it might be better to make it an array or a list depending on how EF Core converts it to SQL (if EF Core is converting the hashset into an array to pass to PGSQL, then the HashSet gives no advantage).